### PR TITLE
Align docstrings to hf guidelines

### PR DESCRIPTION
**Pull request by AndreaFrancis**
See https://github.com/huggingface/datasets-server/pull/2338#issuecomment-1909765283 for previous context.
This non-critical PR tries to align the datasets-server current docstring to HF guidelines (See https://github.com/huggingface/doc-builder for most details).
Changes:
- Remove `:obj:` notation - See internal https://www.notion.so/huggingface2/Documentation-926791574d274a5194cc83699c4a238e
- Remove `Raises the following errors` and `tip` - See https://github.com/huggingface/doc-builder#raises
- Move `Raises` before `Return` for functions - See https://github.com/huggingface/doc-builder#raises
- Optional parameters start and end with `*`- See https://github.com/huggingface/doc-builder#parmeter-typing-and-default-value
- `Attributes` in classes start and end with  `**` - See https://github.com/huggingface/doc-builder#attributes
- Args type start and end with ` - See https://github.com/huggingface/doc-builder#arguments
- Remove `required` from args since it is not needed - See https://github.com/huggingface/doc-builder#arguments
- Remove `Defaults to None` See https://github.com/huggingface/doc-builder#parmeter-typing-and-default-value 
- Remove `Return none`
- Align Returns type https://github.com/huggingface/doc-builder#returns
- Align `Raises`  See https://github.com/huggingface/doc-builder/#internal-link-to-object and https://github.com/huggingface/doc-builder/#raises 
- In workers, align step name in the doc header and first log message

### CODE REVIEW

File: libs/libcommon/src/libcommon/parquet_utils.py [hunk 1]
```diff
@@ -140,14 +140,14 @@ def query(self, offset: int, length: int) -> pa.Table:
...
-            length (int): The number of rows to read.
-
-        Returns:
-            pa.Table: The requested rows.
+            offset (`int`): The first row to read.
+            length (`int`): The number of rows to read.
 
         Raises:
-            TooBigRows: if the arrow data from the parquet row groups is bigger than max_arrow_data_in_memory
+            []`TooBigRows`]: if the arrow data from the parquet row groups is bigger than max_arrow_data_in_memory
```
Comment (severo) [inline libs/libcommon/src/libcommon/parquet_utils.py:147]:
```suggestion
            [`TooBigRows`]: if the arrow data from the parquet row groups is bigger than max_arrow_data_in_memory
```

File: services/admin/src/admin/authentication.py [hunk 2]
```diff
@@ -18,17 +18,17 @@ async def auth_check(
...
-        organization (str | None): the organization name. If None, the dataset is always
+        organization (`str`, *optional*): the organization name. If None, the dataset is always
           authorized.
-        hf_timeout_seconds (float | None): the timeout in seconds for the HTTP request to the external authentication
+        hf_timeout_seconds (`float`, *optional*): the timeout in seconds for the HTTP request to the external authentication
           service.
 
     Returns:
-        None: the user is authorized
+        `Literal[bool]`: the user is authorized
```
Comment (severo) [inline services/admin/src/admin/authentication.py:31]:
```suggestion
        `Literal[True]`: the user is authorized
```

File: services/admin/src/admin/routes/force_refresh.py [hunk 3]
```diff
@@ -61,9 +61,7 @@ async def force_refresh_endpoint(request: Request) -> Response:
                 raise InvalidParameterError(
                     f"Parameter 'priority' should be one of {', '.join(prio.value for prio in Priority)}."
                 )
-            logging.info(
-                f"/force-refresh/{job_type}, dataset={dataset}, config={config}, split={split}, priority={priority}"
-            )
+            logging.info(f"/force-refresh/{job_type}, dataset={dataset}, {config=}, split={split}, {priority=}")
```
Comment (severo) [inline services/admin/src/admin/routes/force_refresh.py:64]:
```suggestion
            logging.info(f"/force-refresh/{job_type}, {dataset=}, {config=}, {split=}, {priority=}")
```

File: services/rows/src/rows/routes/rows.py [hunk 4]
```diff
@@ -72,7 +72,7 @@ async def rows_endpoint(request: Request) -> Response:
                     offset = get_request_parameter_offset(request)
                     length = get_request_parameter_length(request)
                     logging.info(
-                        f"/rows, dataset={dataset}, config={config}, split={split}, offset={offset}, length={length}"
+                        f"/rows, dataset={dataset}, {config=}, split={split}, offset={offset}, length={length}"
```
Comment (severo) [inline services/rows/src/rows/routes/rows.py:75]:
```suggestion
                        f"/rows, {dataset=}, {config=}, {split=}, {offset=}, {length=}"
```

File: services/search/src/search/routes/filter.py [hunk 5]
```diff
@@ -84,7 +84,7 @@ async def filter_endpoint(request: Request) -> Response:
                     offset = get_request_parameter_offset(request)
                     length = get_request_parameter_length(request)
                     logger.info(
-                        f'/filter, dataset={dataset}, config={config}, split={split}, where="{where}",'
+                        f'/filter, dataset={dataset}, {config=}, split={split}, where="{where}",'
                         f" offset={offset}, length={length}"
```
Comment (severo) [inline services/search/src/search/routes/filter.py:88]:
```suggestion
                        f'/filter, {dataset=}, {config=}, {split=}, {where=}, {offset=}, {length=}'
```

File: services/worker/src/worker/job_runners/dataset/size.py [hunk 6]
```diff
@@ -26,20 +26,22 @@
...
     Returns:
         `DatasetSizeResponse`: An object with the sizes_response.
-    Raises the following errors:
-        - [`libcommon.simple_cache.CachedArtifactError`]
-          If the previous step gave an error.
-        - [`libcommon.exceptions.PreviousStepFormatError`]
-            If the content of the previous step has not the expected format
     """
-    logging.info(f"get sizes for dataset={dataset}")
+    logging.info(f"get 'dataset-size' for {dataset}")
```
Comment (severo) [inline services/worker/src/worker/job_runners/dataset/size.py:44]:
```suggestion
    logging.info(f"get 'dataset-size' for {dataset=}")
```

**Review by severo — APPROVED** [review]
Some typos

**Comment by severo** [comment]
Also: surely not worth it, but: do you know if there exists a linter to do this automatically + check quality? I would love not to have to remember all these rules.

File: jobs/cache_maintenance/tests/utils.py [hunk 7]
```diff
@@ -56,25 +56,24 @@ def update_repo_settings(
...
-            `None`.
-    Returns:
-        The HTTP response in json.
-    <Tip>
-    Raises the following errors:
-        - [`~huggingface_hub.utils.RepositoryNotFoundError`]
+            space, `None` or `"model"` if uploading to a model.
+
+    Raises:
+        [~`huggingface_hub.utils.RepositoryNotFoundError`]
```
Comment (severo) [inline jobs/cache_maintenance/tests/utils.py:71]: I think it's the other way, no?

```suggestion
        [`~huggingface_hub.utils.RepositoryNotFoundError`]
```

https://www.notion.so/huggingface2/Documentation-926791574d274a5194cc83699c4a238e

**Comment by severo** [comment]
Also: this project is opensource, but we align the style on an unpublished standard. Should we publish it somewhere?

File: jobs/cache_maintenance/tests/utils.py (hunk 7 above)
Comment (severo) [reply jobs/cache_maintenance/tests/utils.py:71]: sorry, https://www.notion.so/huggingface2/Documentation-926791574d274a5194cc83699c4a238e is not source, it's https://github.com/huggingface/doc-builder

**Comment by severo** [comment]
Also, maybe add a link to https://github.com/huggingface/doc-builder in the CONTRIBUTING.md file

File: libs/libcommon/src/libcommon/parquet_utils.py (hunk 1 above)
Comment (AndreaFrancis) [reply libs/libcommon/src/libcommon/parquet_utils.py:147]: Thanks for catching it!

**Comment by AndreaFrancis** [comment]
> Also: surely not worth it, but: do you know if there exists a linter to do this automatically + check quality? I would love not to have to remember all these rules.

I found these ones: https://pypi.org/project/pydoclint/, https://pypi.org/project/darglint2/ and https://pypi.org/project/flake8-docstrings/. Will evaluate which one could help us with doc validation and if it is possible to have custom docstrings guidelines (since ours is a mix of numpy format and custom HF rules).

**Pull request merged**